textworld/generator/__init__.py

```diff
@@ -113,22 +119,44 @@ def make_world_with(rooms, rng=None):
     return world
 
 
-def make_quest(world, quest_length, rng=None, rules_per_depth=(), backward=False):
-    state = world
-    if hasattr(world, "state"):
-        state = world.state
+def make_quest(world: Union[World, State], options: Optional[GameOptions] = None):
+    state = getattr(world, "state", world)
 
-    rng = g_rng.next() if rng is None else rng
+    if options is None:
+        options = GameOptions()
+
+        # By default, exclude quests finishing with: go, examine, look and inventory.
+        exclude = ["go.*", "examine.*", "look.*", "inventory.*"]
+        options.chaining.rules_per_depth = [options.kb.rules.get_matching(".*", exclude=exclude)]
+        options.chaining.rng = options.rngs['quest']
+
+    chains = []
+    for _ in range(options.nb_parallel_quests):
+        chain = sample_quest(state, options.chaining)
+        if chain is None:
+            msg = "No quest can be generated with the provided options."
+            raise TextworldNoSuchQuestExistError(msg)
+
+        chains.append(chain)
+        state = chain.initial_state  # TODO: the name initial_state doesn't make sense when backward=False
```

Comment:
@tavianator see this comment.

Reply:
Ok, so when `backward` is `False` the `chain.initial_state` is the same as the state passed to [`get_chains`](https://github.com/Microsoft/TextWorld/blob/master/textworld/generator/chaining.py#L480)? That would mean the logic logic in my new `make_quest` is wrong in that case. :O

Reply:
Not exactly the same, since new objects may have been created by chaining.  But it's basically the state passed to `get_chains()` plus the new objects.  Strictly speaking we just take the goal state and run all the actions backwards to get the true initial state.

Reply:
Oh wow. Somehow everytime I'll look at that part of the code, I was reading it like `if self.backward:`. It makes sense now. thx